src/platform/plugins/shared/usage_collection/server/routes/stats/instrument_otel_legacy_metrics.ts

```diff
@@ -0,0 +1,462 @@
+      hostname: string;
+      port: number;
+    };
+  }
+) {
+  const meter = metrics.getMeter('kibana.stats', config.kibanaVersion);
+
+  const collectionIntervalGauge = meter.createGauge('collection_interval_ms', {
+    description:
+      'How often the metrics are calculated in Kibana (in milliseconds). Not necessarily the same as how often OTel ships the metrics (although recommended to have the same value).',
+    // unit: 'ms',
+    valueType: ValueType.INT,
+  });
+
+  const metricsMap = new Map<string, Gauge>();
+
+  combineLatest([metricsServiceSetup.getOpsMetrics$(), overallStatus$, coreStatus$]).subscribe(
+    ([opsMetrics, overallStatus, coreStatus]) => {
+      const attributes = {
+        // Labels (non-numeric signals) coming from the ops metrics
+        'process.pid': opsMetrics.process.pid, // Providing it as an attribute as well as it might be more useful for filtering
+        'kibana.stats.process.event_loop_delay_histogram.from_timestamp':
+          opsMetrics.process.event_loop_delay_histogram.fromTimestamp,
+        'kibana.stats.process.event_loop_delay_histogram.last_updated_at':
+          opsMetrics.process.event_loop_delay_histogram.lastUpdatedAt,
+        'kibana.stats.os.platform': opsMetrics.os.platform,
+        'kibana.stats.os.platform_release': opsMetrics.os.platformRelease,
+        'kibana.stats.os.distro': opsMetrics.os.distro,
+        'kibana.stats.os.distro_release': opsMetrics.os.distroRelease,
+        'kibana.stats.os.cpuacct.control_group': opsMetrics.os.cpuacct?.control_group,
+        'kibana.stats.os.cpu.control_group': opsMetrics.os.cpu?.control_group,
+
+        // From the /api/stats transformations
+        'kibana.uuid': config.uuid,
+        'kibana.name': config.server.name,
+        'kibana.index': config.kibanaIndex,
+        'kibana.host': config.server.hostname,
+        'kibana.locale': i18n.getLocale(),
+        'kibana.transport_address': `${config.server.hostname}:${config.server.port}`,
+        'kibana.version': config.kibanaVersion.replace(SNAPSHOT_REGEX, ''),
+        'kibana.snapshot': SNAPSHOT_REGEX.test(config.kibanaVersion),
+        'kibana.status': ServiceStatusToLegacyState[overallStatus.level.toString()],
+        last_updated: opsMetrics.collected_at.toISOString(),
+
+        // Add fields specific to /api/status (to match the "status" metricset from Metricbeat)
+        'kibana.status.overall.level': overallStatus.level.toString(),
+        'kibana.status.overall.summary': overallStatus.summary,
+        'kibana.status.core.elasticsearch.level': coreStatus.elasticsearch.level.toString(),
+        'kibana.status.core.elasticsearch.summary': coreStatus.elasticsearch.summary,
+        'kibana.status.core.savedObjects.level': coreStatus.savedObjects.level.toString(),
+        'kibana.status.core.savedObjects.summary': coreStatus.savedObjects.summary,
+      };
+
+      collectionIntervalGauge.record(metricsServiceSetup.collectionInterval, attributes);
```

Comment:
I don't have a lot of experience with metrics, but I had thought that high cardinality attribute *values* for metrics would lead to an explosion of time series. Here we have attributes that are UUIDs, timestamps, PIDs. These are high-cardinality values.

Reply:
FMI: are you concerned about the explosion of time series at the OTLP level or at ES level?

For the same instance of Kibana, UUID and PID will be the same for all the records.

Re timestamps, the cardinality will depend on the interval of the Metric Exporter vs. the interval of the OpsMetrics collector: If the Metric Exporter ships metrics every 10s, and the OpsMetrics collector runs every 5s, there will be 2 different timestamps.

Do you think that this is a problem? Happy to remove any attributes that you consider harmful.
For the record, these are the same fields that we report today via Metricbeat (and ES is happy with it so far).